### DISCUSSION TITLE
naniar package and shadow data frame

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#9797] [General]
This relates to the other naniar issue introduced by Lily, see #3711. And to the export of data from CLIMSOFT to R-Instat.
 There are 2 aspects of the data in CLIMSOFT that may not be included in the first round of exporting:
a) Each observation can have a flag associated with it.  This is used to denote T for trace, or E for estimated or D for dubious.  It is possible that the export of the data could include additional columns with thr flag field information.
b) In CLIMSOFT data are entered into a temporary database.  After QC they move to the permanent database.  Export is usually from the permanent database but sometimes it would be useful to include also (or instead) the information from the temporary database.  This is usually for monitoring, where the data may be required even before it is checked.

Naniar has the concept of a shadow data frame.  This is currently to signify meissing or not missing in the main data frame.  But the package documentation discusses the idea that there could (in the future) be more uses for this feature.  Perhaps it could be useful for us, to include the flag field information and also the source, etc.  

So, I suggest that we initially do very simple export from CLIMSOFT and then consider refinements later.

**Comment by Ogik99** [comment]
#5064 
#5108

**Comment by rdstern** [comment]
@lilyclements thanks for moving this here.  I have been back to our problem of multiple missing value codes.  This has led me back to naniar.  It is written by the same person who wrote visdat.  We use that a lot.  I now feel naniar, that isn't currently in R-Instat could provide the answers we need?  I have always been obsessed by the way the multiple values often arise when importing from SPSS, etc. 

This led (through haven I think) to a clever way of including special missing values in R.  But I now think this shadow matrix is just what we need.  I think it is the way to do everyting in a nice potentially visual way.  We now have the data from Tricot, where there is a special missing value.  Then the shadow matrix corresponds to the flag fields in the climatic databases.  And it will accommodate the multiple missing when coming from SAS, Stata and SPSS.

We can start by adding the package.  Then there is also a nice geom, that might go nicely into our scatterplot dialog.
